src/overlays/actors/ovl_En_Bjt/z_en_bjt.c

```diff
@@ -31,49 +58,409 @@ const ActorInit En_Bjt_InitVars = {
+            if (EnBjt_Appear(this)) {
+                this->behaviour++;
+                scriptBranch = 1; // Right item
+            }
+            break;
+
+        case TOILET_HAND_BEHAVIOUR_REWARD_GIVEN:
+            EnBjt_ChangeAnimation(this, TOILET_HAND_ANIM_OPEN_HAND);
+            this->behaviour++;
+            scriptBranch = 1; // Right item
+            break;
+
+        default:
+            break;
+    }
+
+    return scriptBranch;
+}
+
+// Used in Update
+s32 EnBjt_CheckTalk(EnBjt* this, PlayState* play) {
+    s32 ret = false;
+
+    if ((this->stateFlags & 7) && Actor_ProcessTalkRequest(&this->actor, &play->state)) {
+        this->stateFlags |= TOILET_HAND_STATE_TALKING;
+        SubS_UpdateFlags(&this->stateFlags, 0, 7);
+        this->msgEventCallback = EnBjt_ChooseBehaviour;
+        this->behaviour = 0;
+        this->actionFunc = EnBjt_Talk;
+        ret = true;
+    }
+    return ret;
+}
+
+// Used in Update
+s32 EnBjt_ChooseAnimation(EnBjt* this, PlayState* play) {
+    Player* player = GET_PLAYER(play);
+    u16 curTextId = play->msgCtx.currentTextId;
 
-#endif
+    if (player->stateFlags1 & 0x440) { // Talking, show item?
+        this->stateFlags |= TOILET_HAND_STATE_TEXTBOX;
+        if (this->textId != curTextId) {
+            switch (curTextId) {
+                case 0x2949: // Wrong item
+                    EnBjt_ChangeAnimation(this, TOILET_HAND_ANIM_WAGGING_FINGER);
+                    break;
 
-extern ColliderCylinderInit D_80BFDF48;
-extern CollisionCheckInfoInit2 D_80BFDF74;
+                case 0x294A: // Right item
+                    EnBjt_ChangeAnimation(this, TOILET_HAND_ANIM_THUMBS_UP);
+                    break;
 
-extern UNK_TYPE D_06002390;
+                default:
+                    break;
+            }
+        }
+        this->textId = curTextId;
+    } else if (this->stateFlags & TOILET_HAND_STATE_TEXTBOX) {
+        this->stateFlags &= ~TOILET_HAND_STATE_TEXTBOX;
+        EnBjt_ChangeAnimation(this, TOILET_HAND_ANIM_WAITING);
+    }
+    return 0;
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Bjt/func_80BFD2E0.s")
+// Action function, based on msgEvent
+void EnBjt_Talk(EnBjt* this, PlayState* play) {
+    s16 yaw = this->actor.yawTowardsPlayer;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Bjt/func_80BFD30C.s")
+    // TODO: Casting to remove warning for now
+    if (func_8010BF58(&this->actor, play, (s32)sMsgEventScript, this->msgEventCallback, &this->msgEventArg4)) {
+        this->actor.flags &= ~ACTOR_FLAG_100;
+        SubS_UpdateFlags(&this->stateFlags, 3, 7);
+        this->stateFlags &= ~TOILET_HAND_STATE_TALKING;
+        this->msgEventArg4 = 0;
+        this->actionFunc = EnBjt_Wait;
+    } else {
+        Math_ApproachS(&this->actor.shape.rot.y, yaw, 4, 0x2AA8);
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Bjt/func_80BFD3A4.s")
+// Change state based on schedule and stateFlags
+void EnBjt_Wait(EnBjt* this, PlayState* play) {
```

Comment:
I'm curious if we still want to name this `EnBjt_FollowSchedule`. Though ??? doesn't follow the general pattern, it is still close.

Reply:
It's tricky because it uses such a simple form of the script, I feel like it's waiting more than actually doing schedule stuff. Curious what others think: should we stick to the new scheme even for simple ones?

Reply:
Personally I think it would be better to stick to this naming scheme